Cannot install/update NocoDB 0.98.1~ynh1

I had NocoDB installed but the update failled. I uninstalled then tried to install without success.
I tried both: the webadmin and the CLI
YunoHost 11.0.10.1 (stable)
NocoDB 0.98.1~ynh1

It seems to be a dependency issue to typescript/typedoc
I believe that the version on the server is too old.

The following log should help...

# The issue
`
 While resolving: nocodb-sdk@0.98.1
 Found: typescript@4.8.4
 node_modules/typescript
   dev typescript@"^4.0.2" from the root project
   peerOptional typescript@"^3.4.1 || ^4.0.0" from eslint-plugin-functional@3.7.2
   node_modules/eslint-plugin-functional
     dev eslint-plugin-functional@"^3.0.2" from the root project
   2 more (swagger-typescript-api, ts-node)

 Could not resolve dependency:
 dev typedoc@"^0.22.17" from the root project

 Conflicting peer dependency: typescript@4.7.4
 node_modules/typescript
   peer typescript@"4.0.x || 4.1.x || 4.2.x || 4.3.x || 4.4.x || 4.5.x || 4.6.x || 4.7.x" from typedoc@0.22.18
   node_modules/typedoc
     dev typedoc@"^0.22.17" from the root project

 Fix the upstream dependency conflict, or retry
 this command with --force, or --legacy-peer-deps
 to accept an incorrect (and potentially broken) dependency resolution.
 ` 

# The full log
```
args:
  app: nocodb
  force: true
  label: null
  no_remove_on_failure: false
ended_at: 2022-10-18 11:02:40.695024
env:
  YNH_APP_ARG_DOMAIN: nocodb.app.maindomain.tld
  YNH_APP_ARG_IS_PUBLIC: '1'
  YNH_APP_BASEDIR: /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50
  YNH_APP_ID: nocodb
  YNH_APP_INSTANCE_NAME: nocodb
  YNH_APP_INSTANCE_NUMBER: '1'
  YNH_APP_MANIFEST_VERSION: 0.98.1~ynh1
  YNH_ARCH: amd64
error: Une erreur est survenue dans le script d'installation de l'application
interface: cli
operation: app_install
parent: null
related_to:
- - app
  - nocodb
started_at: 2022-10-18 11:01:34.852275
success: false
yunohost_version: 11.0.10.1

============

2022-10-18 12:01:34,858: INFO - Installation de nocodb...
2022-10-18 12:01:35,135: DEBUG - Nothing to update in LDAP
2022-10-18 12:01:35,135: DEBUG - Permission 'nocodb.main' mise à jour
2022-10-18 12:01:35,284: DEBUG - Pour voir le journal de cette opération 'Mise à jour de l'URL associée à l'autorisation 'nocodb'', utilisez la commande 'yunohost log show 20221018-110135-permission_url-nocodb'
2022-10-18 12:01:35,651: DEBUG - The permission database has been resynchronized
2022-10-18 12:01:35,861: DEBUG - La configuration de SSOwat a été regénérée
2022-10-18 12:01:36,023: DEBUG - Executing command '['sh', '-c', '/bin/bash -x "./50-opensondage" nocodb jerome_conte main all_users 7>&1']'
2022-10-18 12:01:36,029: DEBUG - + source /usr/share/yunohost/helpers
2022-10-18 12:01:36,029: DEBUG - +++ set +o
2022-10-18 12:01:36,030: DEBUG - +++ grep xtrace
2022-10-18 12:01:36,031: DEBUG - ++ readonly 'XTRACE_ENABLE=set -o xtrace'
2022-10-18 12:01:36,031: DEBUG - ++ XTRACE_ENABLE='set -o xtrace'
2022-10-18 12:01:36,042: DEBUG - + app=nocodb
2022-10-18 12:01:36,042: DEBUG - + added_users=jerome_conte
2022-10-18 12:01:36,043: DEBUG - + permission=main
2022-10-18 12:01:36,043: DEBUG - + added_groups=all_users
2022-10-18 12:01:36,043: DEBUG - + '[' nocodb == opensondage ']'
2022-10-18 12:01:37,046: DEBUG - Permission 'nocodb.main' créée
2022-10-18 12:01:37,048: DEBUG - Pour voir le journal de cette opération 'Créer permission 'nocodb'', utilisez la commande 'yunohost log show 20221018-110134-permission_create-nocodb'
2022-10-18 12:01:37,090: DEBUG - Executing command '['sh', '-c', '/bin/bash -x "./install"  7>&1']'
2022-10-18 12:01:37,105: DEBUG - + source _common.sh
2022-10-18 12:01:37,106: DEBUG - ++ nodejs_version=16
2022-10-18 12:01:37,106: DEBUG - + source /usr/share/yunohost/helpers
2022-10-18 12:01:37,107: DEBUG - +++ set +o
2022-10-18 12:01:37,108: DEBUG - +++ grep xtrace
2022-10-18 12:01:37,110: DEBUG - ++ readonly 'XTRACE_ENABLE=set -o xtrace'
2022-10-18 12:01:37,110: DEBUG - ++ XTRACE_ENABLE='set -o xtrace'
2022-10-18 12:01:37,125: DEBUG - + ynh_abort_if_errors
2022-10-18 12:01:37,126: DEBUG - + trap ynh_exit_properly EXIT
2022-10-18 12:01:37,126: DEBUG - + domain=nocodb.app.maindomain.tld
2022-10-18 12:01:37,126: DEBUG - + path_url=/
2022-10-18 12:01:37,126: DEBUG - + is_public=1
2022-10-18 12:01:37,126: DEBUG - + app=nocodb
2022-10-18 12:01:37,126: DEBUG - ++ ynh_string_random --length=20
2022-10-18 12:01:37,138: DEBUG - ++ length=20
2022-10-18 12:01:37,139: DEBUG - ++ filter=A-Za-z0-9
2022-10-18 12:01:37,139: DEBUG - ++ dd if=/dev/urandom bs=1 count=1000
2022-10-18 12:01:37,139: DEBUG - ++ sed --quiet 's/\(.\{20\}\).*/\1/p'
2022-10-18 12:01:37,139: DEBUG - ++ tr --complement --delete A-Za-z0-9
2022-10-18 12:01:37,142: DEBUG - + jwt_secret=**********
2022-10-18 12:01:37,200: INFO - [....................] > Validating installation parameters...
2022-10-18 12:01:37,200: DEBUG - + case $YNH_ARCH in
2022-10-18 12:01:37,201: DEBUG - + final_path=/opt/yunohost/nocodb
2022-10-18 12:01:37,201: DEBUG - + test '!' -e /opt/yunohost/nocodb
2022-10-18 12:01:37,201: DEBUG - + ynh_webpath_register --app=nocodb --domain=nocodb.app.maindomain.tld --path_url=/
2022-10-18 12:01:37,247: DEBUG - + yunohost app register-url nocodb nocodb.app.maindomain.tld /
2022-10-18 12:01:39,035: INFO - [+...................] > Storing installation settings...
2022-10-18 12:01:39,036: DEBUG - + ynh_app_setting_set --app=nocodb --key=domain --value=nocodb.app.maindomain.tld
2022-10-18 12:01:39,036: DEBUG - + local _globalapp=nocodb
2022-10-18 12:01:39,083: DEBUG - + app=nocodb
2022-10-18 12:01:39,083: DEBUG - + [[ domain =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:39,083: DEBUG - + ynh_app_setting set nocodb domain nocodb.app.maindomain.tld
2022-10-18 12:01:39,120: DEBUG - + ynh_app_setting_set --app=nocodb --key=path --value=/
2022-10-18 12:01:39,120: DEBUG - + local _globalapp=nocodb
2022-10-18 12:01:39,167: DEBUG - + app=nocodb
2022-10-18 12:01:39,167: DEBUG - + [[ path =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:39,167: DEBUG - + ynh_app_setting set nocodb path /
2022-10-18 12:01:39,202: DEBUG - + ynh_app_setting_set --app=nocodb --key=jwt_secret --value=**********
2022-10-18 12:01:39,202: DEBUG - + local _globalapp=nocodb
2022-10-18 12:01:39,249: DEBUG - + app=nocodb
2022-10-18 12:01:39,249: DEBUG - + [[ jwt_secret =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:39,249: DEBUG - + ynh_app_setting set nocodb jwt_secret **********
2022-10-18 12:01:39,335: INFO - [#+..................] > Finding an available port...
2022-10-18 12:01:39,336: DEBUG - ++ ynh_find_port --port=8095
2022-10-18 12:01:39,344: DEBUG - ++ test -n 8095
2022-10-18 12:01:39,344: DEBUG - ++ ynh_port_available --port=8095
2022-10-18 12:01:39,352: DEBUG - ++ ss --numeric --listening --tcp --udp
2022-10-18 12:01:39,352: DEBUG - ++ awk '{print$5}'
2022-10-18 12:01:39,352: DEBUG - ++ grep --quiet --extended-regexp ':8095$'
2022-10-18 12:01:39,355: DEBUG - ++ grep -q 'port: '\''8095'\''' /etc/yunohost/apps/abantecart/settings.yml /etc/yunohost/apps/ampache/settings.yml /etc/yunohost/apps/civicrm_drupal7/settings.yml /etc/yunohost/apps/diagramsnet/settings.yml /etc/yunohost/apps/dolibarr/settings.yml /etc/yunohost/apps/drupal/settings.yml /etc/yunohost/apps/ethercalc/settings.yml /etc/yunohost/apps/etherpad_mypads/settings.yml /etc/yunohost/apps/keeweb/settings.yml /etc/yunohost/apps/limesurvey/settings.yml /etc/yunohost/apps/navidrome/settings.yml /etc/yunohost/apps/nextcloud/settings.yml /etc/yunohost/apps/nocodb/settings.yml /etc/yunohost/apps/nodered/settings.yml /etc/yunohost/apps/ofbiz/settings.yml /etc/yunohost/apps/opensondage/settings.yml /etc/yunohost/apps/phpboost/settings.yml /etc/yunohost/apps/phpmyadmin/settings.yml /etc/yunohost/apps/roundcube/settings.yml /etc/yunohost/apps/sitemagiccms/settings.yml /etc/yunohost/apps/tiki/settings.yml /etc/yunohost/apps/vaultwarden/settings.yml /etc/yunohost/apps/webmin/settings.yml /etc/yunohost/apps/whitebophir/settings.yml /etc/yunohost/apps/yeswiki/settings.yml
2022-10-18 12:01:39,356: DEBUG - ++ return 0
2022-10-18 12:01:39,357: DEBUG - ++ echo 8095
2022-10-18 12:01:39,357: DEBUG - + port=8095
2022-10-18 12:01:39,357: DEBUG - + ynh_app_setting_set --app=nocodb --key=port --value=8095
2022-10-18 12:01:39,357: DEBUG - + local _globalapp=nocodb
2022-10-18 12:01:39,405: DEBUG - + app=nocodb
2022-10-18 12:01:39,406: DEBUG - + [[ port =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:39,406: DEBUG - + ynh_app_setting set nocodb port 8095
2022-10-18 12:01:39,492: INFO - [##++................] > Installing dependencies...
2022-10-18 12:01:39,492: DEBUG - + ynh_install_nodejs --nodejs_version=16
2022-10-18 12:01:39,500: DEBUG - + mkdir --parents /opt/node_n
2022-10-18 12:01:39,501: DEBUG - + CLEAR_PATH=/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:39,501: DEBUG - ++ echo /opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:39,501: DEBUG - ++ sed s@/usr/local/bin:@@
2022-10-18 12:01:39,503: DEBUG - + PATH=/opt/node_n/bin:/usr/local/sbin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:39,503: DEBUG - + test -x /usr/bin/node
2022-10-18 12:01:39,503: DEBUG - + test -x /usr/bin/npm
2022-10-18 12:01:39,503: DEBUG - + /opt/node_n/bin/n --version
2022-10-18 12:01:39,508: DEBUG - ++ /opt/node_n/bin/n --version
2022-10-18 12:01:39,512: DEBUG - + dpkg --compare-versions v7.5.0 lt 8.2.0
2022-10-18 12:01:39,514: WARNING - dpkg: avertissement: mauvaise syntaxe de la version « v7.5.0 »: le numéro de version ne commence pas par un chiffre
2022-10-18 12:01:39,515: DEBUG - + ynh_replace_string '--match_string=^N_PREFIX=${N_PREFIX-.*}$' '--replace_string=N_PREFIX=${N_PREFIX-/opt/node_n}' --target_file=/opt/node_n/bin/n
2022-10-18 12:01:39,564: DEBUG - + sed --in-place 's@^N_PREFIX=${N_PREFIX-.*}$@N_PREFIX=${N_PREFIX-/opt/node_n}@g' /opt/node_n/bin/n
2022-10-18 12:01:39,565: DEBUG - + PATH=/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:39,565: DEBUG - + test -x /usr/bin/node_n
2022-10-18 12:01:39,565: DEBUG - + test -x /usr/bin/npm_n
2022-10-18 12:01:39,566: DEBUG - ++ uname --machine
2022-10-18 12:01:39,566: DEBUG - + uname=x86_64
2022-10-18 12:01:39,567: DEBUG - + [[ x86_64 =~ aarch64 ]]
2022-10-18 12:01:39,567: DEBUG - + [[ x86_64 =~ arm64 ]]
2022-10-18 12:01:39,567: DEBUG - + n 16
2022-10-18 12:01:40,298: DEBUG -    installed : v16.18.0 (with npm 8.19.2)
2022-10-18 12:01:40,299: DEBUG - ++ sort --version-sort
2022-10-18 12:01:40,299: DEBUG - ++ find /opt/node_n/n/versions/node/16 /opt/node_n/n/versions/node/16.11.0 /opt/node_n/n/versions/node/16.12.0 /opt/node_n/n/versions/node/16.13.0 /opt/node_n/n/versions/node/16.18.0 -maxdepth 0
2022-10-18 12:01:40,299: DEBUG - ++ tail --lines=1
2022-10-18 12:01:40,301: DEBUG - + real_nodejs_version=/opt/node_n/n/versions/node/16.18.0
2022-10-18 12:01:40,301: DEBUG - ++ basename /opt/node_n/n/versions/node/16.18.0
2022-10-18 12:01:40,302: DEBUG - + real_nodejs_version=16.18.0
2022-10-18 12:01:40,302: DEBUG - + '[' '!' -e /opt/node_n/n/versions/node/16 ']'
2022-10-18 12:01:40,303: DEBUG - + echo nocodb:16
2022-10-18 12:01:40,303: DEBUG - + tee --append /opt/node_n/ynh_app_version
2022-10-18 12:01:40,304: DEBUG - nocodb:16
2022-10-18 12:01:40,305: DEBUG - + ynh_app_setting_set --app=nocodb --key=nodejs_version --value=16
2022-10-18 12:01:40,305: DEBUG - + local _globalapp=nocodb
2022-10-18 12:01:40,354: DEBUG - + app=nocodb
2022-10-18 12:01:40,354: DEBUG - + [[ nodejs_version =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:40,354: DEBUG - + ynh_app_setting set nocodb nodejs_version 16
2022-10-18 12:01:40,389: DEBUG - + ynh_cron_upgrade_node
2022-10-18 12:01:40,389: DEBUG - + cat
2022-10-18 12:01:40,390: DEBUG - + chmod +x /opt/node_n/node_update.sh
2022-10-18 12:01:40,391: DEBUG - + cat
2022-10-18 12:01:40,392: DEBUG - + chmod +x /etc/cron.daily/node_update
2022-10-18 12:01:40,393: DEBUG - + ynh_use_nodejs
2022-10-18 12:01:40,394: DEBUG - ++ ynh_app_setting_get --app=nocodb --key=nodejs_version
2022-10-18 12:01:40,394: DEBUG - ++ local _globalapp=nocodb
2022-10-18 12:01:40,417: DEBUG - ++ app=nocodb
2022-10-18 12:01:40,417: DEBUG - ++ [[ nodejs_version =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:40,418: DEBUG - ++ ynh_app_setting get nocodb nodejs_version
2022-10-18 12:01:40,453: DEBUG - + nodejs_version=16
2022-10-18 12:01:40,453: DEBUG - + nodejs_path=/opt/node_n/n/versions/node/16/bin
2022-10-18 12:01:40,453: DEBUG - + shopt -s expand_aliases
2022-10-18 12:01:40,453: DEBUG - + ynh_node=/opt/node_n/n/versions/node/16/bin/node
2022-10-18 12:01:40,453: DEBUG - + alias ynh_node=/opt/node_n/n/versions/node/16/bin/node
2022-10-18 12:01:40,453: DEBUG - + ynh_npm=/opt/node_n/n/versions/node/16/bin/npm
2022-10-18 12:01:40,453: DEBUG - + alias ynh_npm=/opt/node_n/n/versions/node/16/bin/npm
2022-10-18 12:01:40,453: DEBUG - + [[ :/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin: != *\:\/\o\p\t\/\n\o\d\e\_\n\/\n\/\v\e\r\s\i\o\n\s\/\n\o\d\e\/\1\6\/\b\i\n* ]]
2022-10-18 12:01:40,453: DEBUG - + PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:40,453: DEBUG - + node_PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:40,453: DEBUG - + ynh_node_load_PATH=PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:40,453: DEBUG - + ynh_node_load_path=PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:40,507: DEBUG - + ynh_system_user_create --username=nocodb --home_dir=/opt/yunohost/nocodb
2022-10-18 12:01:40,508: INFO - [####+...............] > Configuring system user...
2022-10-18 12:01:40,553: DEBUG - + use_shell=0
2022-10-18 12:01:40,553: DEBUG - + home_dir=/opt/yunohost/nocodb
2022-10-18 12:01:40,553: DEBUG - + groups=
2022-10-18 12:01:40,553: DEBUG - + ynh_system_user_exists nocodb
2022-10-18 12:01:40,560: DEBUG - + getent passwd nocodb
2022-10-18 12:01:40,564: DEBUG - + '[' -n /opt/yunohost/nocodb ']'
2022-10-18 12:01:40,564: DEBUG - + local 'user_home_dir=--home-dir /opt/yunohost/nocodb'
2022-10-18 12:01:40,564: DEBUG - + local 'shell=--shell /usr/sbin/nologin'
2022-10-18 12:01:40,564: DEBUG - + useradd --home-dir /opt/yunohost/nocodb --system --user-group nocodb --shell /usr/sbin/nologin
2022-10-18 12:01:41,184: INFO - [#####+..............] > Creating a MySQL database...
2022-10-18 12:01:41,184: DEBUG - ++ ynh_sanitize_dbid --db_name=nocodb
2022-10-18 12:01:41,194: DEBUG - ++ echo nocodb
2022-10-18 12:01:41,194: DEBUG - + db_name=nocodb
2022-10-18 12:01:41,194: DEBUG - + db_user=nocodb
2022-10-18 12:01:41,194: DEBUG - + ynh_app_setting_set --app=nocodb --key=db_name --value=nocodb
2022-10-18 12:01:41,194: DEBUG - + local _globalapp=nocodb
2022-10-18 12:01:41,250: DEBUG - + app=nocodb
2022-10-18 12:01:41,250: DEBUG - + [[ db_name =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:41,250: DEBUG - + ynh_app_setting set nocodb db_name nocodb
2022-10-18 12:01:41,291: DEBUG - + ynh_mysql_setup_db --db_user=nocodb --db_name=nocodb
2022-10-18 12:01:41,292: DEBUG - + db_pwd=
2022-10-18 12:01:41,326: DEBUG - ++ ynh_string_random
2022-10-18 12:01:41,327: DEBUG - ++ length=24
2022-10-18 12:01:41,327: DEBUG - ++ filter=A-Za-z0-9
2022-10-18 12:01:41,328: DEBUG - ++ dd if=/dev/urandom bs=1 count=1000
2022-10-18 12:01:41,328: DEBUG - ++ tr --complement --delete A-Za-z0-9
2022-10-18 12:01:41,328: DEBUG - ++ sed --quiet 's/\(.\{24\}\).*/\1/p'
2022-10-18 12:01:41,331: DEBUG - + local new_db_pwd=**********
2022-10-18 12:01:41,331: DEBUG - + db_pwd=**********
2022-10-18 12:01:41,331: DEBUG - + ynh_mysql_create_db nocodb nocodb **********
2022-10-18 12:01:41,332: DEBUG - + local db=nocodb
2022-10-18 12:01:41,332: DEBUG - + local 'sql=CREATE DATABASE nocodb;'
2022-10-18 12:01:41,332: DEBUG - + [[ 3 -gt 1 ]]
2022-10-18 12:01:41,332: DEBUG - + sql+=' GRANT ALL PRIVILEGES ON nocodb.* TO '\''nocodb'\''@'\''localhost'\'''
2022-10-18 12:01:41,332: DEBUG - + [[ -n ********** ]]
2022-10-18 12:01:41,332: DEBUG - + sql+=' IDENTIFIED BY '\''**********'\'''
2022-10-18 12:01:41,332: DEBUG - + sql+=' WITH GRANT OPTION;'
2022-10-18 12:01:41,332: DEBUG - + ynh_mysql_execute_as_root '--sql=CREATE DATABASE nocodb; GRANT ALL PRIVILEGES ON nocodb.* TO '\''nocodb'\''@'\''localhost'\'' IDENTIFIED BY '\''**********'\'' WITH GRANT OPTION;'
2022-10-18 12:01:41,344: DEBUG - + database=
2022-10-18 12:01:41,345: DEBUG - + '[' -n '' ']'
2022-10-18 12:01:41,345: DEBUG - + mysql -B ''
2022-10-18 12:01:41,710: DEBUG - + ynh_app_setting_set --app=nocodb --key=mysqlpwd --value=**********
2022-10-18 12:01:41,712: DEBUG - + local _globalapp=nocodb
2022-10-18 12:01:41,772: DEBUG - + app=nocodb
2022-10-18 12:01:41,772: DEBUG - + [[ mysqlpwd =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:41,772: DEBUG - + ynh_app_setting set nocodb mysqlpwd **********
2022-10-18 12:01:41,860: INFO - [######+.............] > Setting up source files...
2022-10-18 12:01:41,861: DEBUG - + ynh_app_setting_set --app=nocodb --key=final_path --value=/opt/yunohost/nocodb
2022-10-18 12:01:41,861: DEBUG - + local _globalapp=nocodb
2022-10-18 12:01:41,915: DEBUG - + app=nocodb
2022-10-18 12:01:41,915: DEBUG - + [[ final_path =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:41,916: DEBUG - + ynh_app_setting set nocodb final_path /opt/yunohost/nocodb
2022-10-18 12:01:41,953: DEBUG - + ynh_setup_source --dest_dir=/opt/yunohost/nocodb
2022-10-18 12:01:41,978: DEBUG - + source_id=app
2022-10-18 12:01:41,979: DEBUG - + keep=
2022-10-18 12:01:41,979: DEBUG - + full_replace=0
2022-10-18 12:01:41,979: DEBUG - + local src_file_path=/var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/app.src
2022-10-18 12:01:41,979: DEBUG - ++ grep SOURCE_URL= /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/app.src
2022-10-18 12:01:41,979: DEBUG - ++ cut --delimiter== --fields=2-
2022-10-18 12:01:41,981: DEBUG - + local src_url=https://github.com/nocodb/nocodb-seed/archive/93c28e2fc6ba7403f147ac8e35c7386f1a9c2194.zip
2022-10-18 12:01:41,981: DEBUG - ++ grep SOURCE_SUM= /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/app.src
2022-10-18 12:01:41,982: DEBUG - ++ cut --delimiter== --fields=2-
2022-10-18 12:01:41,983: DEBUG - + local src_sum=a35d3d491228b4f1cffb3905a9d7fda93957b46d0842febcc90a2c098b5c6744
2022-10-18 12:01:41,984: DEBUG - ++ grep SOURCE_SUM_PRG= /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/app.src
2022-10-18 12:01:41,984: DEBUG - ++ cut --delimiter== --fields=2-
2022-10-18 12:01:41,986: DEBUG - + local src_sumprg=sha256sum
2022-10-18 12:01:41,987: DEBUG - ++ grep SOURCE_FORMAT= /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/app.src
2022-10-18 12:01:41,987: DEBUG - ++ cut --delimiter== --fields=2-
2022-10-18 12:01:41,989: DEBUG - + local src_format=zip
2022-10-18 12:01:41,990: DEBUG - ++ grep SOURCE_EXTRACT= /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/app.src
2022-10-18 12:01:41,990: DEBUG - ++ cut --delimiter== --fields=2-
2022-10-18 12:01:41,992: DEBUG - + local src_extract=true
2022-10-18 12:01:41,992: DEBUG - ++ grep SOURCE_IN_SUBDIR= /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/app.src
2022-10-18 12:01:41,992: DEBUG - ++ cut --delimiter== --fields=2-
2022-10-18 12:01:41,994: DEBUG - + local src_in_subdir=true
2022-10-18 12:01:41,995: DEBUG - ++ grep SOURCE_FILENAME= /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/app.src
2022-10-18 12:01:41,995: DEBUG - ++ cut --delimiter== --fields=2-
2022-10-18 12:01:41,997: DEBUG - + local src_filename=
2022-10-18 12:01:41,997: DEBUG - + src_sumprg=sha256sum
2022-10-18 12:01:41,997: DEBUG - + src_in_subdir=true
2022-10-18 12:01:41,997: DEBUG - + src_format=zip
2022-10-18 12:01:41,997: DEBUG - ++ echo zip
2022-10-18 12:01:41,998: DEBUG - ++ tr '[:upper:]' '[:lower:]'
2022-10-18 12:01:41,999: DEBUG - + src_format=zip
2022-10-18 12:01:41,999: DEBUG - + src_extract=true
2022-10-18 12:01:41,999: DEBUG - + '[' '' = '' ']'
2022-10-18 12:01:41,999: DEBUG - + src_filename=app.zip
2022-10-18 12:01:41,999: DEBUG - + local local_src=/opt/yunohost-apps-src/nocodb/app.zip
2022-10-18 12:01:41,999: DEBUG - + mkdir -p /var/cache/yunohost/download/nocodb/
2022-10-18 12:01:42,000: DEBUG - + src_filename=/var/cache/yunohost/download/nocodb/app.zip
2022-10-18 12:01:42,000: DEBUG - + test -e /opt/yunohost-apps-src/nocodb/app.zip
2022-10-18 12:01:42,001: DEBUG - + '[' -n https://github.com/nocodb/nocodb-seed/archive/93c28e2fc6ba7403f147ac8e35c7386f1a9c2194.zip ']'
2022-10-18 12:01:42,001: DEBUG - ++ wget --tries 3 --no-dns-cache --timeout 900 --no-verbose --output-document=/var/cache/yunohost/download/nocodb/app.zip https://github.com/nocodb/nocodb-seed/archive/93c28e2fc6ba7403f147ac8e35c7386f1a9c2194.zip
2022-10-18 12:01:48,064: DEBUG - + out='2022-10-18 12:01:48 URL:https://codeload.github.com/nocodb/nocodb-seed/zip/93c28e2fc6ba7403f147ac8e35c7386f1a9c2194 [3173] -> "/var/cache/yunohost/download/nocodb/app.zip" [1]'
2022-10-18 12:01:48,065: DEBUG - + sha256sum --check --status
2022-10-18 12:01:48,066: DEBUG - + rm -rf /var/cache/yunohost/files_to_keep_during_setup_source/
2022-10-18 12:01:48,067: DEBUG - + '[' -n '' ']'
2022-10-18 12:01:48,067: DEBUG - + mkdir --parents /opt/yunohost/nocodb
2022-10-18 12:01:48,068: DEBUG - + '[' -n /opt/yunohost/nocodb ']'
2022-10-18 12:01:48,068: DEBUG - + '[' /opt/yunohost/nocodb == /opt/yunohost/nocodb ']'
2022-10-18 12:01:48,068: DEBUG - + _ynh_apply_default_permissions /opt/yunohost/nocodb
2022-10-18 12:01:48,068: DEBUG - + local target=/opt/yunohost/nocodb
2022-10-18 12:01:48,069: DEBUG - ++ jq -r .requirements.yunohost /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/manifest.json
2022-10-18 12:01:48,069: DEBUG - ++ tr -d '>= '
2022-10-18 12:01:48,100: DEBUG - + local ynh_requirement=4.3.0
2022-10-18 12:01:48,100: DEBUG - + '[' -z 4.3.0 ']'
2022-10-18 12:01:48,100: DEBUG - + '[' 4.3.0 == null ']'
2022-10-18 12:01:48,100: DEBUG - + dpkg --compare-versions 4.3.0 ge 4.2
2022-10-18 12:01:48,101: DEBUG - + chmod o-rwx /opt/yunohost/nocodb
2022-10-18 12:01:48,102: DEBUG - + chmod g-w /opt/yunohost/nocodb
2022-10-18 12:01:48,103: DEBUG - + chown -R root:root /opt/yunohost/nocodb
2022-10-18 12:01:48,108: DEBUG - + ynh_system_user_exists nocodb
2022-10-18 12:01:48,116: DEBUG - + getent passwd nocodb
2022-10-18 12:01:48,119: DEBUG - + chown nocodb:nocodb /opt/yunohost/nocodb
2022-10-18 12:01:48,123: DEBUG - + echo /opt/yunohost/nocodb
2022-10-18 12:01:48,123: DEBUG - + grep -q '^/etc/cron'
2022-10-18 12:01:48,124: DEBUG - + true
2022-10-18 12:01:48,125: DEBUG - + '[' zip = zip ']'
2022-10-18 12:01:48,125: DEBUG - + true
2022-10-18 12:01:48,125: DEBUG - ++ mktemp --directory
2022-10-18 12:01:48,126: DEBUG - + local tmp_dir=/tmp/tmp.kLPFgAg53m
2022-10-18 12:01:48,126: DEBUG - + unzip -quo /var/cache/yunohost/download/nocodb/app.zip -d /tmp/tmp.kLPFgAg53m
2022-10-18 12:01:48,128: DEBUG - + cp --archive /tmp/tmp.kLPFgAg53m/nocodb-seed-93c28e2fc6ba7403f147ac8e35c7386f1a9c2194/. /opt/yunohost/nocodb
2022-10-18 12:01:48,129: DEBUG - + ynh_secure_remove --file=/tmp/tmp.kLPFgAg53m
2022-10-18 12:01:48,145: DEBUG - + ynh_secure_remove --file=/var/cache/yunohost/download/nocodb/app.zip
2022-10-18 12:01:48,161: DEBUG - + '[' -d /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/sources/patches/ ']'
2022-10-18 12:01:48,162: DEBUG - + test -e /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/sources/extra_files/app
2022-10-18 12:01:48,162: DEBUG - + '[' -n '' ']'
2022-10-18 12:01:48,162: DEBUG - + rm -rf /var/cache/yunohost/files_to_keep_during_setup_source/
2022-10-18 12:01:48,163: DEBUG - + chmod 750 /opt/yunohost/nocodb
2022-10-18 12:01:48,164: DEBUG - + chmod -R o-rwx /opt/yunohost/nocodb
2022-10-18 12:01:48,165: DEBUG - + chown -R nocodb:www-data /opt/yunohost/nocodb
2022-10-18 12:01:48,230: INFO - [#######+............] > Configuring NGINX web server...
2022-10-18 12:01:48,230: DEBUG - + ynh_add_nginx_config
2022-10-18 12:01:48,230: DEBUG - + local finalnginxconf=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,231: DEBUG - + ynh_add_config --template=nginx.conf --destination=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,257: DEBUG - + '[' -f /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/nginx.conf ']'
2022-10-18 12:01:48,257: DEBUG - + template_path=/var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/nginx.conf
2022-10-18 12:01:48,257: DEBUG - + ynh_backup_if_checksum_is_different --file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,266: DEBUG - + local checksum_setting_name=checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf
2022-10-18 12:01:48,266: DEBUG - ++ ynh_app_setting_get --app=nocodb --key=checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf
2022-10-18 12:01:48,266: DEBUG - ++ local _globalapp=nocodb
2022-10-18 12:01:48,293: DEBUG - ++ app=nocodb
2022-10-18 12:01:48,293: DEBUG - ++ [[ checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:48,293: DEBUG - ++ ynh_app_setting get nocodb checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf
2022-10-18 12:01:48,330: DEBUG - + local checksum_value=
2022-10-18 12:01:48,330: DEBUG - + backup_file_checksum=
2022-10-18 12:01:48,330: DEBUG - + '[' -n '' ']'
2022-10-18 12:01:48,331: DEBUG - + touch /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,331: DEBUG - + chown root:root /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,333: DEBUG - + chmod 640 /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,334: DEBUG - + cp -f /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/conf/nginx.conf /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,335: DEBUG - + _ynh_apply_default_permissions /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,335: DEBUG - + local target=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,336: DEBUG - ++ jq -r .requirements.yunohost /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/manifest.json
2022-10-18 12:01:48,336: DEBUG - ++ tr -d '>= '
2022-10-18 12:01:48,366: DEBUG - + local ynh_requirement=4.3.0
2022-10-18 12:01:48,366: DEBUG - + '[' -z 4.3.0 ']'
2022-10-18 12:01:48,367: DEBUG - + '[' 4.3.0 == null ']'
2022-10-18 12:01:48,367: DEBUG - + dpkg --compare-versions 4.3.0 ge 4.2
2022-10-18 12:01:48,368: DEBUG - + chmod o-rwx /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,368: DEBUG - + chmod g-w /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,369: DEBUG - + chown -R root:root /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,371: DEBUG - + ynh_system_user_exists nocodb
2022-10-18 12:01:48,379: DEBUG - + getent passwd nocodb
2022-10-18 12:01:48,381: DEBUG - + chown nocodb:nocodb /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,382: DEBUG - + echo /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,383: DEBUG - + grep -q '^/etc/cron'
2022-10-18 12:01:48,384: DEBUG - + ynh_replace_vars --file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,392: DEBUG - + test -n /
2022-10-18 12:01:48,392: DEBUG - + local path_url_slash_less=
2022-10-18 12:01:48,392: DEBUG - + ynh_replace_string --match_string=__PATH__/ --replace_string=/ --target_file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,441: DEBUG - + sed --in-place s@__PATH__/@/@g /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,443: DEBUG - + ynh_replace_string --match_string=__PATH__ --replace_string=/ --target_file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,492: DEBUG - + sed --in-place s@__PATH__@/@g /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,493: DEBUG - + test -n nocodb
2022-10-18 12:01:48,493: DEBUG - + ynh_replace_string --match_string=__NAME__ --replace_string=nocodb --target_file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,544: DEBUG - + sed --in-place s@__NAME__@nocodb@g /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,545: DEBUG - + ynh_replace_string --match_string=__NAMETOCHANGE__ --replace_string=nocodb --target_file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,596: DEBUG - + sed --in-place s@__NAMETOCHANGE__@nocodb@g /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,597: DEBUG - + ynh_replace_string --match_string=__USER__ --replace_string=nocodb --target_file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,648: DEBUG - + sed --in-place s@__USER__@nocodb@g /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,653: DEBUG - + test -n /opt/yunohost/nocodb
2022-10-18 12:01:48,653: DEBUG - + ynh_replace_string --match_string=__FINALPATH__ --replace_string=/opt/yunohost/nocodb --target_file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,711: DEBUG - + sed --in-place s@__FINALPATH__@/opt/yunohost/nocodb@g /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,712: DEBUG - + test -n 7.4
2022-10-18 12:01:48,712: DEBUG - + ynh_replace_string --match_string=__PHPVERSION__ --replace_string=7.4 --target_file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,763: DEBUG - + sed --in-place s@__PHPVERSION__@7.4@g /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,764: DEBUG - + test -n PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:48,765: DEBUG - + ynh_replace_string --match_string=__YNH_NODE_LOAD_PATH__ --replace_string=PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin --target_file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,816: DEBUG - + sed --in-place s@__YNH_NODE_LOAD_PATH__@PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin@g /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,818: DEBUG - ++ grep -oP '__[A-Z0-9]+?[A-Z0-9_]*?[A-Z0-9]*?__' /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,818: DEBUG - ++ sort --unique
2022-10-18 12:01:48,819: DEBUG - ++ sed 's@__\([^.]*\)__@\L\1@g'
2022-10-18 12:01:48,821: DEBUG - + uniques_vars=('port')
2022-10-18 12:01:48,821: DEBUG - + local delimit=@
2022-10-18 12:01:48,821: DEBUG - + for one_var in "${uniques_vars[@]}"
2022-10-18 12:01:48,821: DEBUG - + [[ -v port ]]
2022-10-18 12:01:48,821: DEBUG - + match_string=__PORT__
2022-10-18 12:01:48,821: DEBUG - + match_string=__PORT__
2022-10-18 12:01:48,821: DEBUG - + replace_string=8095
2022-10-18 12:01:48,821: DEBUG - + replace_string=8095
2022-10-18 12:01:48,821: DEBUG - + replace_string=8095
2022-10-18 12:01:48,821: DEBUG - + sed --in-place s@__PORT__@8095@g /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,822: DEBUG - + ynh_store_file_checksum --file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,823: DEBUG - + update_only=0
2022-10-18 12:01:48,836: DEBUG - + local checksum_setting_name=checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf
2022-10-18 12:01:48,837: DEBUG - ++ md5sum /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,837: DEBUG - ++ cut '--delimiter= ' --fields=1
2022-10-18 12:01:48,839: DEBUG - + ynh_app_setting_set --app=nocodb --key=checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf --value=62f2a3ea1434149998b891c7e003e2b0
2022-10-18 12:01:48,839: DEBUG - + local _globalapp=nocodb
2022-10-18 12:01:48,891: DEBUG - + app=nocodb
2022-10-18 12:01:48,892: DEBUG - + [[ checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:48,892: DEBUG - + ynh_app_setting set nocodb checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf 62f2a3ea1434149998b891c7e003e2b0
2022-10-18 12:01:48,928: DEBUG - + '[' -n '' ']'
2022-10-18 12:01:48,928: DEBUG - + unset backup_file_checksum
2022-10-18 12:01:48,928: DEBUG - + '[' / '!=' / ']'
2022-10-18 12:01:48,928: DEBUG - + ynh_replace_string '--match_string=^#root_path_only' --replace_string= --target_file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,980: DEBUG - + sed --in-place 's@^#root_path_only@@g' /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,981: DEBUG - + ynh_store_file_checksum --file=/etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,981: DEBUG - + update_only=0
2022-10-18 12:01:48,995: DEBUG - + local checksum_setting_name=checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf
2022-10-18 12:01:48,996: DEBUG - ++ md5sum /etc/nginx/conf.d/nocodb.app.maindomain.tld.d/nocodb.conf
2022-10-18 12:01:48,996: DEBUG - ++ cut '--delimiter= ' --fields=1
2022-10-18 12:01:48,997: DEBUG - + ynh_app_setting_set --app=nocodb --key=checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf --value=62f2a3ea1434149998b891c7e003e2b0
2022-10-18 12:01:48,998: DEBUG - + local _globalapp=nocodb
2022-10-18 12:01:49,048: DEBUG - + app=nocodb
2022-10-18 12:01:49,048: DEBUG - + [[ checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:49,048: DEBUG - + ynh_app_setting set nocodb checksum__etc_nginx_conf.d_nocodb.app.maindomain.tld.d_nocodb.conf 62f2a3ea1434149998b891c7e003e2b0
2022-10-18 12:01:49,083: DEBUG - + '[' -n '' ']'
2022-10-18 12:01:49,084: DEBUG - + unset backup_file_checksum
2022-10-18 12:01:49,084: DEBUG - + ynh_systemd_action --service_name=nginx --action=reload
2022-10-18 12:01:49,150: DEBUG - + service_name=nginx
2022-10-18 12:01:49,150: DEBUG - + action=reload
2022-10-18 12:01:49,150: DEBUG - + line_match=
2022-10-18 12:01:49,150: DEBUG - + length=20
2022-10-18 12:01:49,150: DEBUG - + log_path=/var/log/nginx/nginx.log
2022-10-18 12:01:49,150: DEBUG - + timeout=300
2022-10-18 12:01:49,151: DEBUG - + '[' reload == stop ']'
2022-10-18 12:01:49,151: DEBUG - + '[' reload == reload ']'
2022-10-18 12:01:49,151: DEBUG - + action=reload-or-restart
2022-10-18 12:01:49,151: DEBUG - + systemctl reload-or-restart nginx
2022-10-18 12:01:49,395: DEBUG - + ynh_use_nodejs
2022-10-18 12:01:49,395: INFO - [########++++........] > Installing NocoDB...
2022-10-18 12:01:49,395: DEBUG - ++ ynh_app_setting_get --app=nocodb --key=nodejs_version
2022-10-18 12:01:49,396: DEBUG - ++ local _globalapp=nocodb
2022-10-18 12:01:49,424: DEBUG - ++ app=nocodb
2022-10-18 12:01:49,424: DEBUG - ++ [[ nodejs_version =~ (unprotected|protected|skipped)_ ]]
2022-10-18 12:01:49,424: DEBUG - ++ ynh_app_setting get nocodb nodejs_version
2022-10-18 12:01:49,460: DEBUG - + nodejs_version=16
2022-10-18 12:01:49,460: DEBUG - + nodejs_path=/opt/node_n/n/versions/node/16/bin
2022-10-18 12:01:49,460: DEBUG - + shopt -s expand_aliases
2022-10-18 12:01:49,460: DEBUG - + ynh_node=/opt/node_n/n/versions/node/16/bin/node
2022-10-18 12:01:49,460: DEBUG - + alias ynh_node=/opt/node_n/n/versions/node/16/bin/node
2022-10-18 12:01:49,460: DEBUG - + ynh_npm=/opt/node_n/n/versions/node/16/bin/npm
2022-10-18 12:01:49,460: DEBUG - + alias ynh_npm=/opt/node_n/n/versions/node/16/bin/npm
2022-10-18 12:01:49,460: DEBUG - + [[ :/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin: != *\:\/\o\p\t\/\n\o\d\e\_\n\/\n\/\v\e\r\s\i\o\n\s\/\n\o\d\e\/\1\6\/\b\i\n* ]]
2022-10-18 12:01:49,460: DEBUG - + node_PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:49,461: DEBUG - + ynh_node_load_PATH=PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:49,461: DEBUG - + ynh_node_load_path=PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
2022-10-18 12:01:49,461: DEBUG - + pushd /opt/yunohost/nocodb
2022-10-18 12:01:49,461: DEBUG - /opt/yunohost/nocodb /var/cache/yunohost/app_tmp_work_dirs/app_cct4i_50/scripts
2022-10-18 12:01:49,461: DEBUG - + '[' amd64 = arm64 ']'
2022-10-18 12:01:49,461: DEBUG - + ynh_exec_warn_less ynh_exec_as nocodb PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin /opt/node_n/n/versions/node/16/bin/npm install colors@1.4.0 --save-exact
2022-10-18 12:01:49,461: DEBUG - + [[ 7 -eq 1 ]]
2022-10-18 12:01:49,461: DEBUG - + ynh_exec_as nocodb PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin /opt/node_n/n/versions/node/16/bin/npm install colors@1.4.0 --save-exact
2022-10-18 12:01:49,461: DEBUG - + local user=nocodb
2022-10-18 12:01:49,461: DEBUG - + shift 1
2022-10-18 12:01:49,461: DEBUG - ++ whoami
2022-10-18 12:01:49,462: DEBUG - + [[ nocodb = root ]]
2022-10-18 12:01:49,462: DEBUG - + sudo -u nocodb PATH=/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin /opt/node_n/n/versions/node/16/bin/npm install colors@1.4.0 --save-exact
2022-10-18 12:02:18,745: DEBUG - npm WARN deprecated har-validator@5.1.5: this library is no longer supported
2022-10-18 12:02:19,033: DEBUG - npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
2022-10-18 12:02:19,037: DEBUG - npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
2022-10-18 12:02:19,047: DEBUG - npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
2022-10-18 12:02:19,356: DEBUG - npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
2022-10-18 12:02:20,807: DEBUG - npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
2022-10-18 12:02:21,082: DEBUG - npm WARN deprecated node-pre-gyp@0.11.0: Please upgrade to @mapbox/node-pre-gyp: the non-scoped node-pre-gyp package is deprecated and only the @mapbox scoped package will recieve updates in the future
2022-10-18 12:02:21,824: DEBUG - npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
2022-10-18 12:02:23,025: DEBUG - npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
2022-10-18 12:02:23,627: DEBUG - npm WARN deprecated multer@1.4.4: Multer 1.x is affected by CVE-2022-24434. This is fixed in v1.4.4-lts.1 which drops support for versions of Node.js before 6. Please upgrade to at least Node.js 6 and version 1.4.4-lts.1 of Multer. If you need support for older versions of Node.js, we are open to accepting patches that would fix the CVE on the main 1.x release line, whilst maintaining compatibility with Node.js 0.10.
2022-10-18 12:02:25,756: DEBUG - npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-10-18 12:02:26,133: DEBUG - npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-10-18 12:02:27,556: DEBUG - npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.
2022-10-18 12:02:28,132: DEBUG - npm WARN deprecated axios@0.19.2: Critical security vulnerability fixed in v0.21.1. For more information, see https://github.com/axios/axios/pull/3410
2022-10-18 12:02:38,923: DEBUG - npm notice
2022-10-18 12:02:38,924: DEBUG - npm notice New minor version of npm available! 8.0.0 -> 8.19.2
2022-10-18 12:02:38,924: DEBUG - npm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.19.2>
2022-10-18 12:02:38,924: DEBUG - npm notice Run `npm install -g npm@8.19.2` to update!
2022-10-18 12:02:38,924: DEBUG - npm notice
2022-10-18 12:02:38,925: DEBUG - npm ERR! code 1
2022-10-18 12:02:38,925: DEBUG - npm ERR! path /opt/yunohost/nocodb/node_modules/nocodb-sdk
2022-10-18 12:02:38,926: DEBUG - npm ERR! command failed
2022-10-18 12:02:38,927: DEBUG - npm ERR! command sh -c npm install --package-lock-only --ignore-scripts && npx npm-force-resolutions
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR! code ERESOLVE
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR! ERESOLVE could not resolve
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR!
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR! While resolving: nocodb-sdk@0.98.1
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR! Found: typescript@4.8.4
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR! node_modules/typescript
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR!   dev typescript@"^4.0.2" from the root project
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR!   peerOptional typescript@"^3.4.1 || ^4.0.0" from eslint-plugin-functional@3.7.2
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR!   node_modules/eslint-plugin-functional
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR!     dev eslint-plugin-functional@"^3.0.2" from the root project
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR!   2 more (swagger-typescript-api, ts-node)
2022-10-18 12:02:38,927: DEBUG - npm ERR! npm ERR!
2022-10-18 12:02:38,928: DEBUG - npm ERR! npm ERR! Could not resolve dependency:
2022-10-18 12:02:38,928: DEBUG - npm ERR! npm ERR! dev typedoc@"^0.22.17" from the root project
2022-10-18 12:02:38,928: DEBUG - npm ERR! npm ERR!
2022-10-18 12:02:38,928: DEBUG - npm ERR! npm ERR! Conflicting peer dependency: typescript@4.7.4
2022-10-18 12:02:38,928: DEBUG - npm ERR! npm ERR! node_modules/typescript
2022-10-18 12:02:38,928: DEBUG - npm ERR! npm ERR!   peer typescript@"4.0.x || 4.1.x || 4.2.x || 4.3.x || 4.4.x || 4.5.x || 4.6.x || 4.7.x" from typedoc@0.22.18
2022-10-18 12:02:38,928: DEBUG - npm ERR! npm ERR!   node_modules/typedoc
2022-10-18 12:02:38,928: DEBUG - npm ERR! npm ERR!     dev typedoc@"^0.22.17" from the root project
2022-10-18 12:02:38,928: DEBUG - npm ERR! npm ERR!
2022-10-18 12:02:38,928: DEBUG - npm ERR! npm ERR! Fix the upstream dependency conflict, or retry
2022-10-18 12:02:38,929: DEBUG - npm ERR! npm ERR! this command with --force, or --legacy-peer-deps
2022-10-18 12:02:38,929: DEBUG - npm ERR! npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
2022-10-18 12:02:38,929: DEBUG - npm ERR! npm ERR!
2022-10-18 12:02:38,929: DEBUG - npm ERR! npm ERR! See /opt/yunohost/nocodb/.npm/eresolve-report.txt for a full report.
2022-10-18 12:02:38,929: DEBUG - npm ERR!
2022-10-18 12:02:38,929: DEBUG - npm ERR! npm ERR! A complete log of this run can be found in:
2022-10-18 12:02:38,929: DEBUG - npm ERR! npm ERR!     /opt/yunohost/nocodb/.npm/_logs/2022-10-18T11_02_37_656Z-debug.log
2022-10-18 12:02:39,161: DEBUG - 
2022-10-18 12:02:39,161: DEBUG - npm ERR! A complete log of this run can be found in:
2022-10-18 12:02:39,161: DEBUG - npm ERR!     /opt/yunohost/nocodb/.npm/_logs/2022-10-18T11_02_38_952Z-debug.log
2022-10-18 12:02:39,186: DEBUG - + ynh_exit_properly
2022-10-18 12:02:39,690: DEBUG - + ynh_clean_setup
2022-10-18 12:02:39,690: DEBUG - + ynh_clean_check_starting
2022-10-18 12:02:39,690: DEBUG - + '[' -n '' ']'
2022-10-18 12:02:39,690: DEBUG - + '[' -n '' ']'
2022-10-18 12:02:40,693: ERROR - Impossible d'installer nocodb : Une erreur est survenue dans le script d'installation de l'application
```

--- a/README.md
+++ b/README.md
@@ -27,7 +27,7 @@ NocoDB is an open source NoCode platform that turns any database into a smart sp
 *(from NocoDB's website)*
 
 
-**Shipped version:** 0.98.1~ynh1
+**Shipped version:** 0.98.1~ynh2
 
 **Demo:** https://www.nocodb.com/demos
 

--- a/README_fr.md
+++ b/README_fr.md
@@ -27,7 +27,7 @@ NocoDB is an open source NoCode platform that turns any database into a smart sp
 *(from NocoDB's website)*
 
 
-**Version incluse :** 0.98.1~ynh1
+**Version incluse :** 0.98.1~ynh2
 
 **Démo :** https://www.nocodb.com/demos
 

--- a/manifest.json
+++ b/manifest.json
@@ -6,7 +6,7 @@
         "en": "No-code platform that turns any database into a smart spreadsheet",
         "fr": "Plateform sans code qui transforme toute base de données en un tableur intelligent."
     },
-    "version": "0.98.1~ynh1",
+    "version": "0.98.1~ynh2",
     "url": "https://www.nocodb.com",
     "upstream": {
         "license": "AGPL-3.0-only",

--- a/scripts/install
+++ b/scripts/install
@@ -126,14 +126,11 @@ pushd $final_path
 	if [ $YNH_ARCH = "arm64" ]; then
 		ynh_print_info --message="Retrieving and building OracleDB for ARM64..."
 		ynh_exec_warn_less ynh_exec_as $app git clone --recursive https://github.com/oracle/node-oracledb.git oracledb
-		ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH $ynh_npm install ./oracledb
+		ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH NODE_ENV="production" $ynh_npm install ./oracledb
 		ynh_print_info --message="...Done! Installing NocoDB now..."
 	fi
 
-	# https://github.com/Marak/colors.js/issues/285
-	ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH $ynh_npm install colors@1.4.0 --save-exact
-
-	ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH $ynh_npm install
+	ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH NODE_ENV="production" $ynh_npm install --force
 popd
 
 #=================================================

--- a/scripts/upgrade
+++ b/scripts/upgrade
@@ -114,11 +114,11 @@ pushd $final_path
                 ynh_print_info --message="Retrieving and building OracleDB for ARM64..."
 		ynh_secure_remove oracledb
                 ynh_exec_warn_less ynh_exec_as $app git clone --recursive https://github.com/oracle/node-oracledb.git oracledb
-                ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH $ynh_npm install ./oracledb
+                ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH NODE_ENV="production" $ynh_npm install ./oracledb
                 ynh_print_info --message="...Done! Installing NocoDB now..."
         fi
 
-        ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH $ynh_npm install
+        ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH NODE_ENV="production" $ynh_npm install --force
 popd
 
 #=================================================

